mmv1/products/dialogflowcx/Intent.yaml

```diff
@@ -27,16 +27,41 @@ timeouts: !ruby/object:Api::Timeouts
   update_minutes: 40
 custom_code: !ruby/object:Provider::Terraform::CustomCode
   custom_import: templates/terraform/custom_import/dialogflowcx_intent.go.erb
-  pre_create: templates/terraform/pre_create/dialogflow_set_location.go.erb
+  pre_create: templates/terraform/pre_create/dialogflowcx_set_location_skip_default_obj.go.erb
   pre_update: templates/terraform/pre_create/dialogflow_set_location.go.erb
-  pre_delete: templates/terraform/pre_create/dialogflow_set_location.go.erb
+  pre_delete: templates/terraform/pre_delete/dialogflowcx_set_location_skip_default_obj.go.erb
   pre_read: templates/terraform/pre_create/dialogflow_set_location.go.erb
+virtual_fields:
+  - !ruby/object:Api::Type::Boolean
+    name: is_default_welcome_intent
+    immutable: true
```

Comment:
I did some manual experimenting to check edge cases around this field being immutable:

1. This field can make it easy for a Terraform config to have multiple resource blocks that try to manage a single default object. Normally you'd need to perform explicit import actions to get to that state

    - For example if you have multiple `google_dialogflowcx_intent` resources in a config and set `is_default_welcome_intent` to true on them all then you can cause some chaos 😂 Its equally true for the 3 default objects.
    - Fortunately this would present as the resource constantly having diffs present in plans, so hopefully users could debug this themselves easily enough.

2. If you have a block that's controlling the default start flow and then change `is_default_start_flow` from true to false you get an API error that's due to an flow already existing with that `display_name` value.

Reply:
I think the above is unavoidable, but we could put some warnings in the documentation about the first problem. For the second problem I think it's fine for users to get the API error and self-correct.

For the warnings, we can add some callouts to the docs like this in the description:

```diff
    description: |
      Marks this as the [Default Welcome Intent](https://cloud.google.com/dialogflow/cx/docs/concept/intent#welcome) for an agent. When you create an agent, a Default Welcome Intent is created automatically.
      The Default Welcome Intent cannot be deleted; deleting the `google_dialogflow_cx_intent` resource does nothing to the underlying GCP resources.
+
+      ~> Avoid having multiple `google_dialogflow_cx_intent` resources linked to the same agent with `is_default_negative_intent = true` because they will compete to control a single default Intent resource in GCP.
```

That markdown results in this change in the rendered documentation:

![Screenshot 2023-11-01 at 21 14 58](https://github.com/GoogleCloudPlatform/magic-modules/assets/15078782/d8d93390-6052-46db-9fca-f34ce94ca884)

**Could you please add a warning like this below each of the 3 new fields being added in this PR? Thanks!**

Reply:
TIL the `~>`generates a nice looking warning, I was wondering about that. Thanks, and done.